src/NuGet.Clients/NuGet.Tools/DefaultVSCredentialServiceProvider.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.Common;
+using NuGet.Credentials;
+using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
+using NuGet.ProjectManagement;
+
+namespace NuGetVSExtension
```

Comment:
It'd be better if we'd move the implementation class to `NuGet.PackageManagemetn.VisualStudio` project. Thus it won't have to load `NuGet.Tools.dll` to resolve the MEF import